network.py

```diff
@@ -33,38 +34,38 @@ def add_network_layer(self, layer: Layer) -> None:
 
     def train(self, training_set: pd.DataFrame, visualize: bool = False) -> None:
         current_iteration = 0
-        while not self.__stop_condition_met(current_iteration):
+        mean_error = 1.0
```

Comment:
I think it would be better to initalize mean error with `self.__error_threshold`. It would also prevent from any single epoch not running when error threshold is set to more than 1

Reply:
Good idea